### PR TITLE
Dem/add line contact face

### PR DESCRIPTION
**Pull request by KlausBSautter**
hey I added a 2Node contact condition to the DEM application. I just created this pr so that we can discuss it. @tteschemacher  can you have a look if this will cause problems with what you did for IGA?

### CODE REVIEW

File: applications/DEM_application/custom_strategies/strategies/explicit_solver_strategy.cpp [hunk 1]
```diff
@@ -850,8 +850,10 @@ namespace Kratos {
                 it->CalculateRightHandSide(rhs_cond, r_process_info);
                 DEMWall* p_wall = dynamic_cast<DEMWall*> (&(*it));
                 p_wall->CalculateElasticForces(rhs_cond_elas, r_process_info);
-                array_1d<double, 3> Normal_to_Element;
-                p_wall->CalculateNormal(Normal_to_Element);
+                array_1d<double, 3> Normal_to_Element = ZeroVector(3);
+
+                if (geom.size()>2) p_wall->CalculateNormal(Normal_to_Element);
+
```
Comment (maceligueta) [inline applications/DEM_application/custom_strategies/strategies/explicit_solver_strategy.cpp]: I would put this `if` for all the operations inside the loop. I am afraid that a null normal can bring problems in the following lines.

**Comment by KlausBSautter** [comment]
@maceligueta  I am probably too stupid to find your comment but I cant ...

**Comment by maceligueta** [comment]
My bad! Apparently, once I started the review I could see it from all my computers, however, it needed an extra "Submit review" to make it work. Glad we talked about it! It could have been there for ages!

**Review by maceligueta — APPROVED** [review]

**Comment by KlausBSautter** [comment]
@maceligueta  I am sry but are these all tests to be run?:
If yes everything works and I fixed your request

```
The complete list of benchmarks are included at the end of this message as a quick reference.

========== DEM BENCHMARKING RESULTS ==========

=========== DEM DISCONTINUUM TESTS ===========

==== TSUJI PAPER BENCHMARKS. SLIDING REGIME ==

DEM Benchmark 1: OK!........ Test 1 SUCCESSFUL (error: 0.4, time: 16.05s.)
DEM Benchmark 2: OK!........ Test 2 SUCCESSFUL (error: 0.13, time: 48.08s.)
DEM Benchmark 3: OK!........ Test 3 SUCCESSFUL (error: 0.75, time: 85.19s.)
DEM Benchmark 4: OK!........ Test 4 SUCCESSFUL (error: 4.63, time: 63.16s.)
DEM Benchmark 5: OK!........ Test 5 SUCCESSFUL (error: 1.52, time: 87.79s.)
DEM Benchmark 6: OK!........ Test 6 SUCCESSFUL (error: 3.38, time: 59.88s.)
DEM Benchmark 7: OK!........ Test 7 SUCCESSFUL (error: 0.0, time: 6.46s.)
DEM Benchmark 8: OK!........ Test 8 SUCCESSFUL (error: 4.16, time: 59.77s.)
DEM Benchmark 9: OK!........ Test 9 SUCCESSFUL (error: 0.97, time: 41.63s.)

===== THORNTON PAPER TESTS. FULL REGIME. LINEAR LAW =====

DEM Benchmark 10: OK!........ Test 10 (e=0.25) SUCCESSFUL
DEM Benchmark 10: OK!........ Test 10 (e=0.50) SUCCESSFUL
DEM Benchmark 10: OK!........ Test 10 (e=0.75) SUCCESSFUL
DEM Benchmark 10: OK!........ Test 10 (e=0.90) SUCCESSFUL

==== THORNTON PAPER TESTS. FULL REGIME. HERTZIAN LAW ====

DEM Benchmark 11: OK!........ Test 11 (e=0.25) SUCCESSFUL
DEM Benchmark 11: OK!........ Test 11 (e=0.50) SUCCESSFUL
DEM Benchmark 11: OK!........ Test 11 (e=0.75) SUCCESSFUL
DEM Benchmark 11: OK!........ Test 11 (e=0.90) SUCCESSFUL


==== WENSRICH PAPER TEST. ROLLING FRICTION ====

DEM Benchmark 12: OK!........ Test 12 SUCCESSFUL


======== DE/FE CONTACT BENCHMARKS ==========

DEM Benchmark 13: OK!........ Test 13 SUCCESSFUL
DEM Benchmark 14: OK!........ Test 14 SUCCESSFUL
DEM Benchmark 15: OK!........ Test 15 SUCCESSFUL
DEM Benchmark 16: OK!........ Test 16 SUCCESSFUL
DEM Benchmark 17: OK!........ Test 17 SUCCESSFUL


== BASIC CONTINUUM TESTS ==

DEM Benchmark 20: OK!........ Test 20 SUCCESSFUL
DEM Benchmark 21: OK!........ Test 21 SUCCESSFUL
DEM Benchmark 22: OK!........ Test 22 SUCCESSFUL
DEM Benchmark 23: OK!........ Test 23 SUCCESSFUL
DEM Benchmark 24: OK!........ Test 24 SUCCESSFUL
DEM Benchmark 25: OK!........ Test 25 SUCCESSFUL


===== DISCONTINUUM CLUSTERS TESTS =====

DEM Benchmark 30: OK!........ Test 30 SUCCESSFUL
DEM Benchmark 31: OK!........ Test 31 SUCCESSFUL
DEM Benchmark 32: OK!........ Test 32 SUCCESSFUL
DEM Benchmark 33: OK!........ Test 33 SUCCESSFUL

---------------------------------------------------------------------

List of Benchmarks:

DISCONTINUUM TESTS:
Benchmark 01. Elastic normal impact of two identical spheres
Benchmark 02. Elastic normal impact of a sphere against a rigid plane
Benchmark 03. Impact of a sphere against a rigid plane with different coefficients of restitution
Benchmark 04. Oblique impact of a sphere with a rigid plane with constant velocity module and variable incident angles
Benchmark 05. Oblique impact of a sphere with a rigid plane with constant normal velocity and different tangential velocities
Benchmark 06. Impact of a sphere with a rigid plane with a constant normal velocity and variable angular velocities
Benchmark 07. Impact of two identical spheres with a constant normal velocity and different angular velocities
Benchmark 08. Impact of two differently sized spheres with a constant normal velocity and variable angular velocities
Benchmark 09. Impact of two identical spheres with a constant normal velocity and different coefficients of restitution
Benchmark 10. Oblique impact of a sphere with an elastic plane with constant normal velocity and different angular velocities
Benchmark 11. Oblique impact of a sphere with an elastic plane with constant normal velocity and different angular velocities
Benchmark 12. Sphere rotating over a plane surface with Rolling Friction
Benchmark 13. Impact of a low stiffness sphere against a rigid plane divided in small triangular elements
Benchmark 14. Impact of a low stiffness sphere against a rigid edge divided in small triangular elements
Benchmark 15. Impact of a low stiffness sphere against a rigid vertex divided in small triangular elements
Benchmark 16. Spheres contacting multiple entities (facets, edges and vertices)
Benchmark 17. Sphere sliding on a plane (discretized with triangles and quadrilaterals) with friction

CONTINUUM TESTS:
Benchmark 20. Normal compression of two identical spheres
Benchmark 21. Normal compression of two identical indented spheres
Benchmark 22. Tensile test of two identical spheres
Benchmark 23. Tensile test of two identical indented spheres
Benchmark 24. Shear test of two identical spheres by applying rotation
Benchmark 25. Shear test of two identical spheres by applying rotation and radius expansion

DISCONTINUUM CLUSTERS TESTS:
Benchmark 30. Cylinder cluster with imposed angular velocity in two axis (Velocity Verlet + Zhao scheme)
Benchmark 31. Cylinder cluster with imposed angular velocity in two axis (Symplectic Euler + Runge-Kutta scheme)
Benchmark 32. Fiber cluster bouncing without any damping (Velocity Verlet + Zhao scheme)
Benchmark 33. Fiber cluster bouncing without any damping (Symplectic Euler + Runge-Kutta scheme)

```

**Comment by maceligueta** [comment]
Yes! All tests look ok! Many thanks.

**Pull request merged**